src/dataIntegration/transcriptomics/call_fastcore.m

```diff
@@ -0,0 +1,233 @@
+                else
+                  flipped = false;
+                  singleton = true;
+                end
+            else
+                model.S(:,JiRev) = -model.S(:,JiRev);
+                tmp = model.ub(JiRev);
+                model.ub(JiRev) = -model.lb(JiRev);
+                model.lb(JiRev) = -tmp;
+                flipped = true;
+            end
+        end
+    end
+    
+    toRemove = setdiff(model.rxns,model.rxns(A));
+    tissueModel = removeRxns(model_orig, toRemove);
+    tissueModel = removeNonUsedGenes(tissueModel);
+
+end
+
+function Supp = findSparseMode(J, P, singleton, model, tol, scaling)
+% Finds a mode that contains as many reactions from J and as few from P
+% Returns its support, or [] if no reaction from J can get flux above tol
+%* Vlassis, Pacheco, Sauter (2014). Fast reconstruction of compact
+%context-specific metbolic network models. PLoS Comput. Biol. 10, e1003424.
+
+    Supp = [];
+    if isempty(J) 
+      return;
+    end
+
+    if singleton
+      V = LP7(J(1), model, tol);
+    else
+      V = LP7(J, model, tol);
+    end
+
+    K = intersect(J, find(V >= 0.99*tol));   
+    if isempty(K) 
+      return;
+    end
+
+    V = LP9(K, P, model, tol, scaling);
+    Supp = find(abs(V) >= 0.99*tol);
+end
+
+function V = LP7(J, model, tol) %% TO DO, check that it works for the different available solver in cobra, if not fix the potential solver 
+% CPLEX implementation of LP-7 for input set J (see FASTCORE paper)
+% (c) Nikos Vlassis, Maria Pires Pacheco, Thomas Sauter, 2013
+%     LCSB / LSRU, University of Luxembourg
+
+    nj = numel(J);
+    [m,n] = size(model.S);
+
+    % x = [v;z]
+
+    % objective
+    f = -[zeros(1,n), ones(1,nj)];
+
+    % equalities
+    Aeq = [model.S, sparse(m,nj)];
+    beq = zeros(m,1);
+
+    % inequalities
+    Ij = sparse(nj,n); 
+    Ij(sub2ind(size(Ij),(1:nj)',J(:))) = -1;
+    Aineq = sparse([Ij, speye(nj)]);
+    bineq = zeros(nj,1);
+
+    % bounds
+    lb = [model.lb; zeros(nj,1)];
+    ub = [model.ub; ones(nj,1)*tol];
+
+    % Set up problem
+    LPproblem.A = [Aeq;Aineq];
+    LPproblem.b = [beq;bineq];
+    LPproblem.c = f;
+    LPproblem.lb = lb;
+    LPproblem.ub = ub;
+    LPproblem.osense = 1;
+    LPproblem.csense(1:m,1) = 'E';
+    LPproblem.csense(m+1:length(bineq)+m,1) = 'L';
+    
+    sol = solveCobraLP(LPproblem);
+    if sol.stat == 1
+        x = sol.full;    
+        %x = cplexlp(f,Aineq,bineq,Aeq,beq,lb,ub);
+        V = x(1:n);
+    else
+        V = zeros(n,1);
+    end
+end
+
+function V = LP9(K, P, model, tol, scaling)%% TO DO, check that it works for the different available solver in cobra, if not fix the potential solver 
+% CPLEX implementation of LP-9 for input sets K, P (see FASTCORE paper)
+%* Vlassis, Pacheco, Sauter (2014). Fast reconstruction of compact
+%context-specific metbolic network models. PLoS Comput. Biol. 10, e1003424.
+
+    scalingfactor = scaling; %FIX: used to be 1e5, but since tol is smaller, different value

```

Comment:
In the orginal call of fastcore, this scaling factor is fixed to 1e5 (see LP9 function), we just introduced it as a parameter bcz it is influenced by tolerance (named epsilon in original code)